### PR TITLE
fix: [ANDROAPP-3503] Open form/event initial screen when setting date to schedule event

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3503)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/events/ScheduledEventActivity.kt [hunk 1]
```diff
@@ -292,37 +296,48 @@ class ScheduledEventActivity : ActivityGlobalAbstract(), ScheduledEventContract.
...
+        bundle.putSerializable(
+            Constants.EVENT_PERIOD_TYPE,
+            stage.periodType()
+        )
+        bundle.putString(Constants.PROGRAM_STAGE_UID, stage.uid())
+        bundle.putInt(
+            Constants.EVENT_SCHEDULE_INTERVAL,
+            stage.standardInterval() ?: 0
+        )
+        intent.putExtras(bundle)
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/events/ScheduledEventActivity.kt:326]: I think you can use the extension from ActivityExtension @Balcan

File: app/src/main/java/org/dhis2/usescases/events/ScheduledEventPresenterImpl.kt [hunk 2]
```diff
@@ -112,4 +104,17 @@ class ScheduledEventPresenterImpl(
...
+            .blockingGet()
+        val stage = d2.programModule().programStages().uid(event.programStage())
+            .blockingGet()
+        val program = d2.programModule().programs().uid(stage.program()?.uid())
+            .blockingGet()
+        val hasCoordinates = stage.featureType() != null && stage.featureType() != FeatureType.NONE
+        val hasNonDefaultCatCombo = d2.categoryModule().categoryCombos()
+            .uid(program.categoryComboUid()).blockingGet().isDefault != true
+        return hasCoordinates || hasNonDefaultCatCombo
+    }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/events/ScheduledEventPresenterImpl.kt:119]: does it make sense to move it to SDK ???

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/events/ScheduledEventPresenterImpl.kt:119]: Could be yes, i'll move it to to the logic class

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/events/ScheduledEventPresenterImpl.kt:119]: great ! can you move it to the dhis2 logic package? huge thanks !

**Review by mmmateos — APPROVED** [review]

**Review by JaimeToca — APPROVED** [review]

**Pull request merged**